src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/data-broker-profiles/welcome-to-premium/page.tsx

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Link from "next/link";
+import buttonStyles from "../../../../../../../../components/server/button.module.scss";
+import styles from "./welcomeToPremium.module.scss";
+import { getL10n } from "../../../../../../../../functions/server/l10n";
+import { getLatestOnerepScan } from "../../../../../../../../../db/tables/onerep_scans";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../../../../../../../api/utils/auth";
+import { getOnerepProfileId } from "../../../../../../../../../db/tables/subscribers";
+import { redirect } from "next/navigation";
+import { WelcomeToPremiumDonutChart } from "../../../../../../../../components/client/Chart";
+import { getSubscriberBreaches } from "../../../../../../../../functions/server/getUserBreaches";
+import { dashboardSummary } from "../../../../../../../../functions/server/dashboard";
+
+export default async function WelcomeToPremium() {
+  const l10n = getL10n();
+  const session = await getServerSession(authOptions);
+
+  if (!session?.user?.subscriber?.id) {
+    redirect("/redesign/user/dashboard/");
+  }
+
+  const result = await getOnerepProfileId(session.user.subscriber.id);
+  const profileId = result[0]["onerep_profile_id"] as number;
+  const scanResult = await getLatestOnerepScan(profileId);
+  const scanResultItems = scanResult?.onerep_scan_results?.data ?? [];
+  const countOfDataBrokerProfiles = scanResultItems.length;
+  const subBreaches = await getSubscriberBreaches(session.user);
+  const summary = dashboardSummary(scanResultItems, subBreaches);
+  const totalExposures = summary.totalExposures;
+  const exposureReduction = Math.round(
```

Comment:
@codemist I think I'll need to refactor this to a util function, so your `manual removal` screen and reuse

Reply:
Refactored to own function